src/main/java/org/ow2/proactive/connector/iaas/rest/InfrastructureRest.java

```diff
@@ -67,9 +68,12 @@ public Response getInfrastructure(@PathParam("infrastructureId") String infrastr
     @DELETE
     @Path("/{infrastructureId}")
     @Produces("application/json")
-    public Response deleteInfrastructureById(@PathParam("infrastructureId") String infrastructureId) {
+    public Response deleteInfrastructureById(@PathParam("infrastructureId") String infrastructureId,
+            @QueryParam("deleteInstances") Boolean deleteInstances) {
         Optional.ofNullable(infrastructureService.getInfrastructure(infrastructureId))
-                .ifPresent(infrastructure -> infrastructureService.deleteInfrastructure(infrastructure));
+                .ifPresent(infrastructure -> infrastructureService.deleteInfrastructure(infrastructure,
```

Comment:
can we have a separate method in the infrastructureService to delete the instances instead of propagating the boolean and the if downstream in the hierarchy ?